### PR TITLE
Maria marz phonebook test

### PR DESCRIPTION
**Pull request by MariaMarz**

### CODE REVIEW

File: src/test/java/LoginTest.java [hunk 1]
```diff
@@ -16,6 +17,12 @@ public void successLogin() throws InterruptedException {
         ContactsPage contactsPage = new ContactsPage(app.driver);
         sleep(1000);
```
Comment (lkazarin) [inline src/test/java/LoginTest.java:18]: why sleep is here?

File: src/test/java/LoginTest.java [hunk 2]
```diff
@@ -16,6 +17,12 @@ public void successLogin() throws InterruptedException {
...
         assertTrue(contactsPage.contactsPageIsDisplayed());
```
Comment (lkazarin) [inline src/test/java/LoginTest.java:19]: are you using Unit or ng?

File: src/test/java/AddNewContactTest.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+public class AddNewContactTest extends BaseTest {
+
+
+    @Test
+    public void successAddNewContact() throws InterruptedException {
+        LoginHelper loginHelper = new LoginHelper(app.driver);
+        loginHelper.loginTestHelper();
+        ContactsPage contactsPage = new ContactsPage(app.driver);
+        sleep(2000);
```
Comment (lkazarin) [inline src/test/java/AddNewContactTest.java:15]: I don’t think so about “sleep"

File: src/main/java/LoginHelper.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+
+    public void loginTestHelper() throws InterruptedException {
+        LoginPage loginPage = new LoginPage(driver);
+        loginPage.enterEmail("test@gmail.com");
+        loginPage.enterPassword("test@gmail.com");
+        loginPage.clickOnLoginButton();
+        ContactsPage contactsPage = new ContactsPage(driver);
+        sleep(1000);
+        assertTrue(contactsPage.contactsPageIsDisplayed());
+        contactsPage.clickButtonAddNewContact();
```
Comment (lkazarin) [inline src/main/java/LoginHelper.java:20]: In LoginHelper there should be no addition of a contact

File: src/main/java/LoginHelper.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        assertTrue(contactsPage.modalWindowIsDisplayed());
```
Comment (lkazarin) [inline src/main/java/LoginHelper.java:21]: same

File: src/test/java/AddNewContactTest.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+        contactsPage.clickButtonAddNewContact();
+        assertTrue(contactsPage.modalWindowIsDisplayed());
+        AddContactDialog addContactDialog = new AddContactDialog(app.driver);
+        addContactDialog.enterFirstName("Maria");
```
Comment (lkazarin) [inline src/test/java/AddNewContactTest.java:19]: Here you can use variables for first name, last name and description, but not necessarily.

File: src/test/java/AddNewContactTest.java [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+        addContactDialog.enterLastName("Marz");
+        addContactDialog.enterAbout("123");
+        addContactDialog.clickButtonSave();
+        ContactInfoPage contactInfoPage = new ContactInfoPage(app.driver);
+        sleep(2000);
+        assertEquals(contactInfoPage.firstNameIsCorrect(), "Maria");
+        assertEquals(contactInfoPage.lastNameIsCorrect(), "Marz");
+        assertEquals(contactInfoPage.descriptionIsCorrect(), "123");
+        contactInfoPage.clickButtonEdit();
+        contactInfoPage.editAllInputFields("Maria9", "Marz9", "1239");
```
Comment (lkazarin) [inline src/test/java/AddNewContactTest.java:29]: cool!

File: src/test/java/AddNewContactTest.java [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+        contactInfoPage.clickButtonSave();
+        contactInfoPage.clickButtonContact();
+        contactsPage.clickSearchInputField();
+        contactsPage.enterNameInputSearchField("Maria9");
+        assertEquals(contactsPage.contactIsCorrect(), "MariaMaria9 MarzMarz9");
+        contactsPage.clickToButtonDelete();
+        DeleteContactPage deleteContactPage = new DeleteContactPage(app.driver);
```
Comment (lkazarin) [inline src/test/java/AddNewContactTest.java:36]: not Page, but Dialog

File: src/test/java/AddNewContactTest.java [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+        ContactInfoPage contactInfoPage = new ContactInfoPage(app.driver);
+        sleep(2000);
+        assertEquals(contactInfoPage.firstNameIsCorrect(), "Maria");
+        assertEquals(contactInfoPage.lastNameIsCorrect(), "Marz");
+        assertEquals(contactInfoPage.descriptionIsCorrect(), "123");
+        contactInfoPage.clickButtonEdit();
+        contactInfoPage.editAllInputFields("Maria9", "Marz9", "1239");
+        contactInfoPage.clickButtonSave();
+        contactInfoPage.clickButtonContact();
+        contactsPage.clickSearchInputField();
```
Comment (lkazarin) [inline src/test/java/AddNewContactTest.java:32]: Unnecessary action: click in the search field, although you can immediately start writing a search query.

File: src/test/java/AddNewContactTest.java [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+        contactsPage.enterNameInputSearchField("Maria9");
+        assertEquals(contactsPage.contactIsCorrect(), "MariaMaria9 MarzMarz9");
```
Comment (lkazarin) [inline src/test/java/AddNewContactTest.java:34]: I see that when editing contact data, new names were added, not replaced the old names.

Comment (lkazarin) [reply src/test/java/AddNewContactTest.java:34]: In addition, you added "Maria9" when editing, but you check with the name “MariaMaria9"

File: src/main/java/ContactInfoPage.java [hunk 11]
```diff
@@ -7,17 +8,82 @@ public ContactInfoPage(WebDriver driver) {
...
+    public void clickButtonEdit(){
+        try{
+            editButton.click();
+        } catch (ElementClickInterceptedException e){
+            e.getStackTrace();
+        }
+
+    }
+
+    public void enterFirstNameEdit(String firstNameEditValue){
```
Comment (lkazarin) [inline src/main/java/ContactInfoPage.java:63]: I think so better: editFirstName

File: src/main/java/ContactInfoPage.java [hunk 12]
```diff
@@ -7,17 +8,82 @@ public ContactInfoPage(WebDriver driver) {
...
+        firstNameEdit.sendKeys(firstNameEditValue);
```
Comment (lkazarin) [inline src/main/java/ContactInfoPage.java:64]: editFirstName

File: src/main/java/ContactInfoPage.java [hunk 13]
```diff
@@ -7,17 +8,82 @@ public ContactInfoPage(WebDriver driver) {
...
+    }
+
+    public void enterLastNameEdit(String lastNameEditValue){
```
Comment (lkazarin) [inline src/main/java/ContactInfoPage.java:67]: editLastName

File: src/main/java/ContactInfoPage.java [hunk 14]
```diff
@@ -7,17 +8,82 @@ public ContactInfoPage(WebDriver driver) {
...
+        lastNameEdit.sendKeys(lastNameEditValue);
```
Comment (lkazarin) [inline src/main/java/ContactInfoPage.java:68]: editLastName

File: src/main/java/ContactInfoPage.java [hunk 15]
```diff
@@ -7,17 +8,82 @@ public ContactInfoPage(WebDriver driver) {
...
+    }
+
+    public void enterDescriptionEdit(String descriptionEditValue){
```
Comment (lkazarin) [inline src/main/java/ContactInfoPage.java:71]: editDescription

File: src/main/java/ContactInfoPage.java [hunk 16]
```diff
@@ -7,17 +8,82 @@ public ContactInfoPage(WebDriver driver) {
...
+        descriptionEdit.sendKeys(descriptionEditValue);
```
Comment (lkazarin) [inline src/main/java/ContactInfoPage.java:72]: editDescription

File: src/main/java/ContactInfoPage.java (hunk 11 above)
Comment (lkazarin) [reply src/main/java/ContactInfoPage.java:63]: In addition, when editing fields, it is necessary to delete the old data in the field.

For example,

`editFirstName.clear();`

File: src/main/java/ContactInfoPage.java [hunk 17]
```diff
@@ -7,17 +8,82 @@ public ContactInfoPage(WebDriver driver) {
...
+    }
+    public void editAllInputFields(String firstNameEditValue, String lastNameEditValue, String descriptionEditValue){
```
Comment (lkazarin) [inline src/main/java/ContactInfoPage.java:74]: cool!

File: src/main/java/ContactInfoPage.java [hunk 18]
```diff
@@ -7,17 +8,82 @@ public ContactInfoPage(WebDriver driver) {
...
+        enterFirstNameEdit(firstNameEditValue);
+        enterLastNameEdit(lastNameEditValue);
+        enterDescriptionEdit(descriptionEditValue);
+    }
+
+    public void clickButtonSave(){
```
Comment (lkazarin) [inline src/main/java/ContactInfoPage.java:80]: `clickSaveButton` would be better

File: src/main/java/ContactsPage.java [hunk 19]
```diff
@@ -22,13 +23,21 @@ public class ContactsPage extends BasePage {
...
     @FindBy(css = "b")
     private WebElement foundContact;
     @FindBy(css = "img[src='/assets/icons/trash.svg']")
     private WebElement deleteButton;
 
+    @FindBy(id = "add-contact-modal")
+    private WebElement modalWindowAddContact;
+
+    @FindBy(xpath = "//*[@id=\"contacts-list\"]/div/app-contact-item/div/button[1]")
+    private WebElement fromContactsList;
```
Comment (lkazarin) [inline src/main/java/ContactsPage.java:37]: I don’t understand what do you mean `fromContactList`, please be more precise in the names.

For example:
`contactCardFields`

File: src/test/java/AddNewContactTest.java (hunk 10 above)
Comment (lkazarin) [inline src/test/java/AddNewContactTest.java:34]: There is no check of the contents of the description field (`1239`)

File: src/main/java/ContactsPage.java [hunk 20]
```diff
@@ -39,4 +48,43 @@ public boolean contactsPageIsDisplayed() {
...
+    public void enterFirstName(String firstNameValue){
+
+        firstNameInputField.sendKeys(firstNameValue);
+    }
+
+    public void enterLastName(String lastNameValue){
+        lastNameInputField.sendKeys(lastNameValue);
+    }
+
+    public void clickSearchInputField(){
```
Comment (lkazarin) [inline src/main/java/ContactsPage.java:75]: As I have already commented, this is not necessary action.

File: src/main/java/ContactsPage.java [hunk 21]
```diff
@@ -39,4 +48,43 @@ public boolean contactsPageIsDisplayed() {
...
+
+    public void enterNameInputSearchField(String expectedText){
+        searchInputForm.sendKeys(expectedText);
+    }
+
+    public String contactIsCorrect(){
+         return fromContactsList.getText();
+    }
+
+    public void clickToButtonDelete(){
```
Comment (lkazarin) [inline src/main/java/ContactsPage.java:87]: `clickDeleteButton`

File: src/main/java/DeleteContactPage.java [hunk 22]
```diff
@@ -0,0 +1,30 @@
...
+public class DeleteContactPage extends BasePage{
+    public DeleteContactPage(WebDriver driver) {
+        super(driver);
+    }
+
+    @FindBy(id = "check-box-remove-contact")
+    private WebElement checkBox;
+
+    @FindBy(id = "submit-remove")
+    private WebElement buttonDeleteElem;
```
Comment (lkazarin) [inline src/main/java/DeleteContactPage.java:14]: I don't think it was worth cutting a few characters

File: src/main/java/DeleteContactPage.java [hunk 23]
```diff
@@ -0,0 +1,30 @@
...
+
+    @FindBy(xpath = "//*[@class='alert text-center alert-warning']")
+    private WebElement alertText;
+
+    public void clickToCheckBox(){
```
Comment (lkazarin) [inline src/main/java/DeleteContactPage.java:19]: `clickCheckbox`

File: src/main/java/DeleteContactPage.java [hunk 24]
```diff
@@ -0,0 +1,30 @@
...
+        checkBox.click();
```
Comment (lkazarin) [inline src/main/java/DeleteContactPage.java:20]: `checkbox`

File: src/main/java/DeleteContactPage.java [hunk 25]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    public void clickButtonDeleteElement(){
```
Comment (lkazarin) [inline src/main/java/DeleteContactPage.java:23]: `clickDeleteButtonElement`

**Review by lkazarin — APPROVED** [review]

**Review by AlexgorAlexgor — APPROVED** [review]

File: src/test/java/LoginTest.java (hunk 2 above)
Comment (MariaMarz) [reply src/test/java/LoginTest.java:19]: I use Unit

File: src/test/java/LoginTest.java (hunk 1 above)
Comment (MariaMarz) [reply src/test/java/LoginTest.java:18]: I need change method sleep to isDisplayed.

File: src/test/java/AddNewContactTest.java (hunk 3 above)
Comment (MariaMarz) [reply src/test/java/AddNewContactTest.java:15]: I have answered up

File: src/main/java/LoginHelper.java (hunk 4 above)
Comment (MariaMarz) [reply src/main/java/LoginHelper.java:20]: Yes, you are right. I fix this

File: src/main/java/LoginHelper.java (hunk 5 above)
Comment (MariaMarz) [reply src/main/java/LoginHelper.java:21]: what is same?

File: src/test/java/AddNewContactTest.java (hunk 6 above)
Comment (MariaMarz) [reply src/test/java/AddNewContactTest.java:19]: I prefer this variation

File: src/test/java/AddNewContactTest.java (hunk 7 above)
Comment (MariaMarz) [reply src/test/java/AddNewContactTest.java:29]: Thanks

File: src/main/java/ContactsPage.java (hunk 21 above)
Comment (MariaMarz) [reply src/main/java/ContactsPage.java:87]: I fix this

File: src/main/java/DeleteContactPage.java (hunk 22 above)
Comment (MariaMarz) [reply src/main/java/DeleteContactPage.java:14]: I fix it

File: src/main/java/DeleteContactPage.java (hunk 23 above)
Comment (MariaMarz) [reply src/main/java/DeleteContactPage.java:19]: There are two ways to write clickOnCheckBox, clickCheckBox

File: src/main/java/DeleteContactPage.java (hunk 24 above)
Comment (MariaMarz) [reply src/main/java/DeleteContactPage.java:20]: ok

File: src/main/java/DeleteContactPage.java (hunk 25 above)
Comment (MariaMarz) [reply src/main/java/DeleteContactPage.java:23]: ok

File: src/test/java/AddNewContactTest.java (hunk 8 above)
Comment (MariaMarz) [reply src/test/java/AddNewContactTest.java:36]: I fix it

File: src/test/java/AddNewContactTest.java (hunk 9 above)
Comment (MariaMarz) [reply src/test/java/AddNewContactTest.java:32]: Yes, you are right

File: src/test/java/AddNewContactTest.java (hunk 10 above)
Comment (MariaMarz) [reply src/test/java/AddNewContactTest.java:34]: Yes

File: src/main/java/ContactInfoPage.java (hunk 11 above)
Comment (MariaMarz) [reply src/main/java/ContactInfoPage.java:63]: I fix it

File: src/main/java/ContactInfoPage.java (hunk 12 above)
Comment (MariaMarz) [reply src/main/java/ContactInfoPage.java:64]: Ok

File: src/main/java/ContactInfoPage.java (hunk 13 above)
Comment (MariaMarz) [reply src/main/java/ContactInfoPage.java:67]: I fix it

File: src/main/java/ContactInfoPage.java (hunk 14 above)
Comment (MariaMarz) [reply src/main/java/ContactInfoPage.java:68]: Ok

File: src/test/java/LoginTest.java (hunk 1 above)
Comment (lkazarin) [reply src/test/java/LoginTest.java:18]: Doesn’t it work without `sleep`?

File: src/main/java/ContactInfoPage.java (hunk 15 above)
Comment (MariaMarz) [reply src/main/java/ContactInfoPage.java:71]: Ok

File: src/main/java/ContactInfoPage.java (hunk 16 above)
Comment (MariaMarz) [reply src/main/java/ContactInfoPage.java:72]: Ok

File: src/main/java/ContactInfoPage.java (hunk 17 above)
Comment (MariaMarz) [reply src/main/java/ContactInfoPage.java:74]: Thanks

File: src/main/java/ContactInfoPage.java (hunk 18 above)
Comment (MariaMarz) [reply src/main/java/ContactInfoPage.java:80]: I fix it

File: src/main/java/ContactsPage.java (hunk 19 above)
Comment (MariaMarz) [reply src/main/java/ContactsPage.java:37]: I fix it

File: src/test/java/AddNewContactTest.java (hunk 10 above)
Comment (MariaMarz) [reply src/test/java/AddNewContactTest.java:34]: I add

File: src/main/java/ContactsPage.java (hunk 20 above)
Comment (MariaMarz) [reply src/main/java/ContactsPage.java:75]: Ok

File: src/test/java/AddNewContactTest.java (hunk 6 above)
Comment (lkazarin) [reply src/test/java/AddNewContactTest.java:19]: I think this may be required when scaling the autotest.

**Pull request merged**